### ISSUE TITLE
Avalonia - TabControl integration

### ISSUE THREAD

**Issue by antoningar** [NorthwoodsSoftware/GoDiagram#7]
Hello,

I would like to use GoDiagrams with Avalonia TabControls.
One TabItem of my TabControl is a GoDiagram like this one [sample](https://github.com/NorthwoodsSoftware/GoDiagram/blob/b7918721cbd105a2320723452487b7efeb01fd53/Samples/SharedSamples/Samples/BeatPaths).

When I open for the first time the TabItem, diagram is displayed.
When I open an other TabItem and I come back to the Diagram's TabItem, app crash.

Here is a quick video : 
![video](https://github.com/NorthwoodsSoftware/GoDiagram/assets/126385140/751d3a83-7ac3-4b37-9c56-8517e342166b)

The full code is available [here](https://github.com/antoningar/AvaloniaSample/tree/main).

The stacktrace : 
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Northwoods.Go.Avalonia.DiagramControl.OnLoaded(RoutedEventArgs e)
   at Avalonia.Controls.Control.OnLoadedCore()
   at Avalonia.Controls.Control.<>c.<.cctor>b__69_0()
   at Avalonia.Threading.DispatcherOperation.InvokeCore()
   at Avalonia.Threading.DispatcherOperation.Execute()
   at Avalonia.Threading.Dispatcher.ExecuteJob(DispatcherOperation job)
   at Avalonia.Threading.Dispatcher.ExecuteJobsCore(Boolean fromExplicitBackgroundProcessingCallback)
   at Avalonia.Threading.Dispatcher.Signaled()
   at Avalonia.X11.X11PlatformThreading.CheckSignaled()
   at Avalonia.X11.X11PlatformThreading.RunLoop(CancellationToken cancellationToken)
   at Avalonia.Threading.DispatcherFrame.Run(IControlledDispatcherImpl impl)
   at Avalonia.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, Action`1 lifetimeBuilder)
   at AvaloniaSample.Program.Main(String[] args) in Program.cs:line 13
```

No idea where the problem come from but if you have any hint for a fix or a workaround for this use case it would be cool.

**Comment by jonchardy** [comment]
Thanks for reporting. We're working on this.

**Comment by jonchardy** [comment]
We have a fix for this. We'll probably release a new version later this week.

**Comment by antoningar** [comment]
Thank you for your reactivity and this quick good news !

**Comment by jonchardy** [comment]
We just released 10.0.10, which now performs diagram cleanup when the TopLevel control is closed rather than on unload. Tabs should now work as expected.

**Comment by antoningar** [comment]
Working well, thank you !
